### PR TITLE
Add draft grammar for KQL 1.0.0

### PR DESCRIPTION
**Pull request by larsgw**
Might need to be more restrictive.

### CODE REVIEW

**Review by zkat — APPROVED** [review]

**Comment by zkat** [comment]
LGTM, but I'll let others take a look too in case I missed something. I'll also need to update the 2.0 grammar sometime soon after this lands..

**Comment by larsgw** [comment]
I actually want to change some of the whitespace rules now that I look at it. It's missing whitespace after `=>`, I think whitespace should be allowed in more places, and I want to allow newlines in all/most whitespace if that's okay with you.

**Comment by zkat** [comment]
I'm fine with that

**Comment by larsgw** [comment]
Should this be a breaking change? Or just make it non-normative until a new major release?

**Comment by larsgw** [comment]
I tried to split the patch but everything got sent at once so the commit message is a bit misleading, but I:

  - added support for newlines to any whitespace
  - added some places where whitespace is allowed (around most if not all operators)
  - added the two accessors unique to mappings (`props()`, `values()`)
  - added a notice that this grammar is not normative for now

**Comment by zkat** [comment]
I think we should just make KQL 2.0.0 and be done with it, instead of trying to incrementally improve 1.0, but that's just me. wdyt?

**Comment by larsgw** [comment]
That's fine with me too but I think that should be after/together with KDL 2.0.0, not before (given the pending changes to identifiers). I guess KQL 2.0.0 would also include the new selector operators? I'm excited for those.

**Comment by zkat** [comment]
Can you update this PR to target `kdl-v2` instead, then? You'll have to rebase, but idk if you'll run into conflicts.

**Comment by larsgw** [comment]
Nevermind, that didn't really work I think. I'll rebase locally and force-push.

**Comment by larsgw** [comment]
Then I've removed the part of the grammar concerning the map operator too, since that was removed.

File: QUERY-SPEC.md
Comment (larsgw) [reply QUERY-SPEC.md:112]: Good point, updated

**Pull request merged**
